test/unit/app/helpers/admin/republishing_helper_test.rb

```diff
@@ -47,6 +47,33 @@ class Admin::RepublishingHelperTest < ActionView::TestCase
 
     assert_equal first_link, expected_link
   end
+
+  test "#republishing_index_bulk_republishing_rows creates a link to the 'new' path for the 'all_by_type' section" do
+    all_by_type_link = republishing_index_bulk_republishing_rows.flatten.find { |column|
+                         column[:text].include?("all-by-type")
+                       }[:text]
+
+    expected_link = '<a id="all-by-type" class="govuk-link" href="/government/admin/republishing/bulk/by-type/new">Republish <span class="govuk-visually-hidden">all by type</span></a>'
+
+    assert_equal all_by_type_link, expected_link
+  end
+
+  test "#republishable_content_types_select_options creates select options from republishable_content_types" do
+    # we need to eager load here to ensure we have all the models
+    Rails.application.eager_load!
+
+    options = republishable_content_types_select_options
+
+    assert_includes options, {
+      text: "CallForEvidence",
+      value: "call-for-evidence",
+    }
+
+    assert_includes options, {
+      text: "Contact",
+      value: "contact",
+    }
+  end
```

Comment:
I opted against this for a couple of reasons:

1. I didn't want to manually write out every select option manually in the test (although I can if you feel it's important to check every type is included) and the option for looping over a list would have involved doing exactly the same as the implementation which also didn't feel right i.e we'd have to do this to get the expected values:
```
     republishable_content_types.map do |type|
        {
          text: type,
          value: type.underscore.dasherize,
        }
    end
```

2. Stubbing a `republishing_helper` method inside the tests for class itself didn't feel right to me. We _could_ stub its constituent parts like `ApplicationRecord.subclasses[...]` and `Edition.descendants[...]` but that didn't feel great either.

I think our tests are good enough for the feature here as we're testing that one of each option is present in the list.

Reply:
1. I was more meaning stub it to be an array of say two or three types, then just write out the full array of three hashes and `assert_equal` on that
2. I did think about that when writing this comment, but you already do this for the `BulkRepublisher` tests: https://github.com/alphagov/whitehall/pull/9122/files#diff-60fd99a4438aa942f0d060eeb263e1a56843bb86050f4aa55e7d2f907edc225fR207-R208

Reply:
Ah sorry my mistake, I thought `republishable_content_types` was a method in the `BulkRepublisher` rather than an included helper module's method (too many similarly-named methods!), which we confusingly have to stub on the `BulkRepublisher` itself. Yep, I'll sort that.

Reply:
It's stubbing the methods here:
1. https://github.com/alphagov/whitehall/blob/5d79b12529e362bd98f997756966a52048fdb787/app/helpers/admin/republishing_helper.rb#L92
2. https://github.com/alphagov/whitehall/blob/5d79b12529e362bd98f997756966a52048fdb787/app/helpers/admin/republishing_helper.rb#L57

but we have to stub them on the `BulkRepublisher` class as it's an included module, unless there's another way to stub a module method?

Reply:
Ohhh! I suppose because of the way inclusion works, they're inserted into the `BulkRepublisher`'s method list, so it would probably be complicated/not possible to stub them via the `RepublishingHelper` itself?